ViAn/Video/overlay.cpp

```diff
@@ -51,25 +51,38 @@ void Overlay::set_showing_overlay(bool value) {
 /**
  * @brief Overlay::set_tool
  * Sets the overlay tool's shape.
- * If the tool is the text-tool the user is prompted to wnter a text.
- * @param s
+ * If the tool is the text-tool the user is prompted
+ * to enter a text and a font scale.
+ * @param s New tool to be set.
  */
 void Overlay::set_tool(SHAPES s) {
+    current_shape = s;
+
+    // If the text option is choosen, a string and size will be entered by the user.
     if (s == TEXT) {
-        bool ok;
-        current_string = QInputDialog::getText(NULL, "Text chooser", "Enter a text:", QLineEdit::Normal, QString(), &ok);
-        if (!ok || current_string.isEmpty()) {
-            // Cancelled or empty field.
-            return;
+        std::string input_string = current_string.toStdString();
+        float input_font_scale = current_font_scale;
+        CustomDialog dialog("Choose text", NULL);
+        dialog.addLabel("Enter values:");
+        dialog.addLineEdit ("Text:", &input_string, "Enter a text that can then be used to draw on the overlay.");
+        dialog.addDblSpinBoxF("Font scale:", 0.5, 5.0, &input_font_scale, 1, 0.1,
```

Comment:
Magical constants, if deemed unncessesary to make them defined constants, atleast leave a comment explaining why they are what they are